src/components/Table/Table.jsx

```diff
@@ -1,31 +1,203 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import TableHeaderCell from '../TableHeaderCell/TableHeaderCell';
+import Button from '../Button/Button';
+import SubmitRow from '../SubmitRow/SubmitRow';
+import Pagination from '../Pagination/Pagination';
+import { sortRows } from '../../helpers';
 import styles from './Table.module.css';
 
-const students = [
-  { firstName: 'John', secondName: 'Johnson', birthYear: 1978 },
-  { firstName: 'John1', secondName: 'Johnson1', birthYear: 1988 },
-  { firstName: 'John2', secondName: 'Johnson2', birthYear: 1908 },
-  { firstName: 'John3', secondName: 'Johnson3', birthYear: 1998 },
-  { firstName: 'John4', secondName: 'Johnson4', birthYear: 1990 },
-];
-
 class Table extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      students,
+    const firstFieldName = Object.keys(this.props.data[0])[0];
+    const initialState = {
       page: 1,
-      itemPerPage: 2,
+      rowsPerPage: 4,
+      sortFieldName: firstFieldName,
+      sortDirectionAsc: true,
+      students: sortRows(this.props.data, firstFieldName, true),
     };
+    this.state = initialState;
   }
 
+  handleSelect = (event) => {
+    event.persist();
+    this.setState(() => ({
+      page: 1,
+      rowsPerPage: +event.target.value,
+    }));
+  };
+
+  handleDeleteClick = (index) => {
+    this.setState((state) => {
+      const students = state.students.filter((row, i) => i !== index);
+      return { students };
+    });
+  };
+
+  handleSubmitRow = (row) => {
+    const { firstName, secondName, birthYear } = row;
+    if (!firstName || !secondName || !birthYear) {
+      // eslint-disable-next-line no-undef, no-alert
+      alert('Please, fill in the input fields.');
+      return;
+    }
+    this.setState((state) => ({ students: [...state.students, row] }));
+  };
+
+  handleNextClick = () => {
+    this.setState((state) => {
+      const numOfRows = state.students.length;
+      if (Math.ceil(numOfRows / state.rowsPerPage) > state.page) {
+        return { page: state.page + 1 };
+      }
+      return {};
```

Comment:
??

Reply:
Here i wanted to update the state or do nothing depending on the condition in the if statement. So here i'm updating the state or return an empty object which won't update the state.